cupy/_core/_kernel.pyx

```diff
@@ -1638,33 +1622,15 @@ cdef class _Ops:
 
             # check for overflow in operands. Consider `np.uint8(1) + 300`.
             # Per NEP 50 this raises OverflowError because 300 overflows uint8.
-            # We can only check it after the output dtype is known, which is
-            # op.out_types here.
-            out_type = op.out_types[0]
-            if issubclass(out_type, numpy.integer):
-                iinfo = numpy.iinfo(out_type)
-                for i in range(len(in_args)):
-                    arg = in_args[i]
-                    typ = arg.dtype.type
-
-                    if weaks[i] and issubclass(typ, numpy.integer):
-                        if issubclass(typ, numpy.signedinteger):
-
-                            if issubclass(out_type, numpy.signedinteger):
-                                ok = (<int64_t>iinfo.min <=
-                                      <int64_t>arg <=
-                                      <int64_t>iinfo.max)
-                            else:
-                                ok = ((arg >= 0) and
-                                      (<uint64_t>arg <= <uint64_t>iinfo.max))
-
-                        elif issubclass(typ, numpy.unsignedinteger):
-                            # arg >= 0
-                            ok = <uint64_t>arg <= <uint64_t>iinfo.max
-
-                        if not ok:
-                            raise OverflowError(f"Python integer {arg} out of "
-                                                f"bounds for {out_type}.")
+            # We can only check it after the operand types are known
+            for i in range(len(in_args)):
+                if weaks[i] is not int:
+                    continue
+                # Note: For simplicity, check even if `in_type` is e.g. float
+                integer_argument = int(in_args[i])
```

Comment:
whoa. I thought I knew how cython works. That it generates correct int/uint C casts is... TIL something :-).

Reply:
Not sure, I wanted to pass a Python integer to the NumPy coercion because I wasn't quite sure if NumPy would cast it otherwise (with integer overflow).
It does insert correct casts (with some tricky details), but we are not casting to C types here.